Dockerfile

```
@@ -0,0 +1,22 @@
+FROM python as base
+
+# Perform common operations, dependency installation etc...
+RUN pip install poetry
+RUN pip install flask
+COPY pyproject.toml poetry.lock ./
```

Although there's no particular harm here in copying directly without setting a working directory, it does mean there's a risk that we replace standard linux folders during the copy, and it makes it a little harder to navigate within the container. This is also why your hot reloading isn't working, because the code is being mirrored in a separate location (see below):
```suggestion
WORKDIR /app
COPY pyproject.toml poetry.lock ./
```